src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,269 @@
+package duke.util;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringJoiner;
+
+import duke.Duke;
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+
+/**
+ * The class that parses user input and returns parsed objects.
+ */
+public class Parser {
+
+    /**
+     * The method takes a user input and parse out all necessary parts needed
+     * to execute user command and finally returns a Command object that can be executed
+     * @param input A line of user input
+     * @return A Command object which can then be called to execute user command
+     * @throws DukeException When the input is empty String or the command is invalid
+     */
+    public static Command parseCommand(String input) throws DukeException {
+        String[] splited = input.strip().split(" +");
```

Comment:
Plural form should be used on names representing a collection of objects. Perhaps `splitedStrings` would be a better name?